custom/icds_core/view_utils.py

```diff
@@ -26,5 +29,22 @@ def is_icds_cas_project(domain):
     return IS_ICDS_ENVIRONMENT and domain == ICDS_DOMAIN
 
 
+def check_authorization(domain, user, master_app_id):
+    if (
+        settings.SERVER_ENVIRONMENT == 'icds-staging'
+        and toggles.ROLE_WEBAPPS_PERMISSIONS.enabled(user.username)
+    ):
+        try:
+            role = user.get_role(domain)
+        except DomainMembershipError:
+            return HttpResponse(_('User is not a member of this project'), status=404)
+        else:
+            if (
+                not master_app_id
```

Comment:
I would remove this check and put the responsibility on the caller to check if master_app_id is falsy